core/src/main/scala/org/apache/spark/util/ThreadUtils.scala

```diff
@@ -304,6 +305,20 @@ private[spark] object ThreadUtils {
   }
   // scalastyle:on awaitresult
 
+  @throws(classOf[SparkException])
+  def awaitResult[T](future: JFuture[T], atMost: Duration): T = {
+    try {
+      atMost match {
+        case Duration.Inf => future.get()
+        case _ => future.get(atMost._1, atMost._2)
+      }
+    } catch {
```

Comment:
shall we follow the other `awaitResult` and add
```
case e: SparkFatalException =>
        throw e.throwable
```

Reply:
Done